File: src/main/java/gargoyle/netsquares/util/SwingProperties.java
package gargoyle.netsquares.util;

import gargoyle.netsquares.prop.base.BaseRWProperty;

import javax.swing.AbstractButton;

public final class SwingProperties {
    private SwingProperties() {
    }

    public static void bind(BaseRWProperty<Boolean> property, AbstractButton item) {
        item.setSelected(property.get());
        item.addActionListener(e -> property.set(item.isSelected()));
        property.addPropertyListener((prop, oldValue, newValue) -> item.setSelected(newValue));
    }
}

File: src/main/kotlin/gargoyle/netsquares/util/SwingProperties.kt
package gargoyle.netsquares.util

import gargoyle.netsquares.prop.base.BaseRWProperty
import javax.swing.AbstractButton

object SwingProperties {
    fun bind(property: BaseRWProperty<Boolean>, item: AbstractButton) {
        item.isSelected = property.get()
        item.addActionListener { property.set(item.isSelected) }
        property.addPropertyListener { _: BaseRWProperty<Boolean>, _: Boolean, newValue: Boolean ->
            item.isSelected = newValue
        }
    }
}
